src/tool/xmeta/cpp/models/method_model.h

```diff
@@ -5,30 +5,57 @@
 #include <vector>
 
 #include "base_model.h"
+#include "compilation_unit.h"
 #include "formal_parameter_model.h"
-#include "model_ref.h"
-#include "model_types.h"
 
 namespace xlang::xmeta
 {
-    struct method_semantics
+    struct method_modifier
     {
         bool is_protected = false;
         bool is_static = false;
+        bool is_overridable = false;
     };
 
     struct method_model : base_model
     {
         method_model() = delete;
-        method_model(std::string_view const& id, size_t decl_line, 
+        method_model(std::string_view const& id, 
+                size_t decl_line, 
                 std::string_view const& assembly_name, 
                 std::optional<type_ref>&& return_type, 
                 std::vector<formal_parameter_model>&& formal_params,
-                method_semantics const& sem) :
+                method_modifier const& sem,
+                method_association const& assoc) :
             base_model{ id, decl_line, assembly_name },
             m_formal_parameters{ std::move(formal_params) },
             m_return_type{ std::move(return_type) },
-            m_semantic{ sem },
+            m_modifier{ sem },
+            m_association{ assoc },
+            m_implemented_method_ref{ "" }
+        { }
+
+        method_model(std::string_view const& id,
+                size_t decl_line,
+                std::string_view const& assembly_name,
+                std::optional<type_ref>&& return_type,
```

Comment:
you're missing a way of encoding that the return type or any of the parameter types are arrays